src/com/visualization/piechart/piechart.js

```diff
@@ -0,0 +1,69 @@
+import {h, Component} 				from 'preact/preact';
+
+import PieSegment					from 'com/visualization/piechart/piesegment';
+
+export default class PieChart extends Component {
+
+	constructor( props ) {
+		super(props);
+
+    }
+
+    // Converts an array of number to an array of percentages
+    convertToPercentage ( values ) {
+        // calculate total
+        let total = 0;
+        values.forEach( ( v ) => ( total += v ) );
+
+        // scale values so there sum is 100%
+        let percentages = values.map( ( v ) => ( (100 * v) / total ) );
+
+        return percentages;
+    }
+
+    render( props ) {
+
+        if ( !(props && props['lables'] && props['values'])) {
+            console.warn('PieChart was created with invalid props', props);
+            return <div> No Data! </div>;
+        }
+
+        let lables = props.lables;
+        let values = props.values;
+
+        let percentages = this.convertToPercentage(values);
+
+        let Segments = [];
+        let Ledgend = [];
+
+        let offset = 0;
+        for (var i = 0; i < percentages.length; i++) {
+
+            let color = 1 + ( i % 6 );
+            let ledgendclass = "shape-circle ledgend_color_"+color;
```

Comment:
I think it would be nicer to use the global class name formatting function `cN`

```
cN("shape-circle", "legend-color-"+color, props.legendClass);
```
since then ready to be extended if we want per components optionally passing in extra definitions

Reply:
Yeah, cN is new and slowly replacing all class merging fancyness. If you Google the classNames library/function, it's that, but I abbreviated it to cN.

Reply:
Changed to cN implementation.